templates/shortcodes/mermaid.html

```diff
@@ -0,0 +1,36 @@
+{% if body %}  {# inline diagram #}
+    {% set data = body %}
+{% else %}     {# load from file #}
+    {% if section %}
+        {% set base_path="content" ~ section.path %}
+    {% else %}
+        {% set base_path="content" ~ page.path %}
+    {% endif %}
+    {% set data = load_data(path=base_path ~ path) %}
+{% endif%}
+<div style="overflow:auto;">
+<pre class="mermaid pre" style="width:{%if width %}{{width}}{% else %}100%{% endif %};">
+{{ data }}
+</pre>
+</div>
+
+<!-- when this javascript is loaded and initialized it will look for divs with the class mermaid.
+     it will replace the content of those divs with the generated svg -->
+<script type="module">
+ // sourced from https://unpkg.com/mermaid@9/dist/mermaid.esm.min.mjs
```

Comment:
This should be in the commit body since the code can be replaced and new versions could be used.